core/src/main/java/org/pentaho/platform/util/ActionUtil.java

```diff
@@ -89,6 +90,50 @@
   }
 
   /**
+   * Removes the provided {@code key} from the {@code params} {@link Map} , if it exists, as well as its
+   * "equivalent", as defined within the {@code KEY_MAP}. If the provided {@code key} is a key within the {@code
+   * KEY_MAP}, its "equivalent" is the value. If the provided {@code key} is a value within the {@code KEY_MAP}, its
+   * "equivalent" is the key.<br><br>
+   * Example: Given a {@code KEY_MAP} that looks like this:<br>
+   *   <ul><li>firstName=first.name</li>
+   *   <li>lastName=last.name</li>
+   *   <li>addr=address</li>
+   *   <li>person.age=age</li></ul>
+   * a call to {@code removeKeyFromMap( map, "firstName")} will attempt to remove the "firstName" and "first.name" keys
+   * from the provided map. Likewise, a call to  {@code removeKeyFromMap( map, "last.name")} will attempt to remove the
+   * "last.name" and "lastName" keys from the provided map.
+   *
+   * @param params the {@link Map} from which keys are being removed
+   * @param key the {@code String} key being removed from the map
+   */
+  public static void removeKeyFromMap( final Map<String, ?> params, final String key ) {
+    if ( params == null || key == null ) {
+      logger.debug( "Map or key are null, cannot remove." );
+      return;
+    }
+    // remove the key itself
+    params.remove( key );
+
+    // also remove the keys equivalent in the KEY_MAP, whether its a key or a value itself
+    if ( KEY_MAP.keySet().contains( key ) ) {
+      params.remove( KEY_MAP.get( key ) );
+    } else if ( KEY_MAP.values().contains( key ) ) {
```

Comment:
Isn't this if statement wasteful? We are just repeating the same thing twice?

Reply:
no - first we are checking if 'key' is a key with KEY_MAP, then we are checking if it's a value within KEY_MAP. Maybe I'm not understanding the question, we can discuss offline

Reply:
What I meant was:  values().contains() effectively goes over the map. So in case of the key existing we go over the map twice. If it does not exist we effectively go over the map once anyway through the values().contains() ... theoreticaqlly we can just iterate.... but this is minor I guess, ...

Reply:
So as an optimization, we could iterate over entries and check the key and value of each entry for a match - is this what you have in mind?

Reply:
Yes. But again, this is minor, please decide.